code/datums/weather/weather_types/rain_storm.dm

```diff
@@ -0,0 +1,153 @@
+/datum/weather/rain_storm
+	name = "rain"
+	desc = "Heavy thunderstorms rain down below, drenching anyone caught in it."
+
+	telegraph_message = span_danger("Thunder rumbles far above. You hear droplets drumming against the canopy.")
+	telegraph_overlay = "rain_low"
+	telegraph_duration = 30 SECONDS
+
+	weather_message = span_userdanger("<i>Rain pours down around you!</i>")
+	weather_overlay = "rain_high"
+
+	end_message = span_bolddanger("The downpour gradually slows to a light shower.")
+	end_overlay = "rain_low"
+	end_duration = 30 SECONDS
+
+	weather_duration_lower = 3 MINUTES
+	weather_duration_upper = 5 MINUTES
+
+	weather_color = null
+	thunder_color = null
+
+	area_type = /area
+	target_trait = ZTRAIT_RAINSTORM
+	immunity_type = TRAIT_RAINSTORM_IMMUNE
+	probability = 0
+	turf_weather_chance = 0.01
+	turf_thunder_chance = THUNDER_CHANCE_AVERAGE
+
+	weather_flags = (WEATHER_TURFS | WEATHER_MOBS | WEATHER_THUNDER | WEATHER_BAROMETER | WEATHER_NOTIFICATION)
+
+	/// A weighted list of possible reagents that will rain down from the sky.
+	/// Only one of these will be selected to be used as the reagent
+	var/list/whitelist_weather_reagents = list(/datum/reagent/water)
+	/// A list of reagents that are forbidden from being selected when there is no
+	/// whitelist and the reagents are randomized
+	var/list/blacklist_weather_reagents = list()
+	/// The selected reagent that will be rained down
+	var/datum/reagent/rain_reagent
+
+/datum/weather/rain_storm/New(z_levels, area_override, weather_flags_override, thunder_chance_override, datum/reagent/custom_reagent)
+	..()
+
+	if(IS_WEATHER_AESTHETIC(weather_flags))
+		return
+
+	var/reagent_id
```

Comment:
are we checking for anything here or just the turf effects flag

Reply:
If the weather doesn't process on turfs, mobs, or thunder then no need to setup reagent for our weather event. (since nothing is gonna happen) Although doing it this way disables any selected reagent from being used to color, which might be something we want to keep for only aesthetics.

Reply:
Removed the early return via https://github.com/tgstation/tgstation/pull/89550/commits/f69b183951435c1bede58ba7e3de91c3982c0c42